config/routes.rb

```diff
@@ -3,4 +3,12 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  resources :dishes, only: [:show] do
+    resources :dish_ingredients, only: [:create]
+  end
+
+  resources :chefs, only: [:show] do
+    resources :ingredients, only: [:index]
```

Comment:
This is a great RESTful route. What it creates is: `/chefs/:chef_id/ingredients` which is lovely but it directed it to `ingredients#index` which is okay but not the best option. You could do `controller: "chef_ingredients"` or `controller: "chefs/ingredients"`